scripts/commit-msg

```diff
@@ -0,0 +1,87 @@
+#!/usr/bin/env ruby
+
+# This script is intended to be used under repo_dir/.git/hooks to check
+# each git commit message against OMR contributing guidelines. The script
+# should be executable and it will be invoked when a 'git commit' command 
+# is issued.
+
+message_file = ARGV[0]
+message = File.read(message_file)
+
+# remove git generated messages
+commit = ""
+message.each_line do |line|
+  if (line.match(/^(?!#).*$/))
+    commit.concat(line)
+  end
+end
+
+title, body = commit.split("\n", 2)
+
+# check commit title length
+title_len = title.length
+#puts "Title is #{title}, length is #{title_len}"
+
+if (title_len > 0 && title_len < 50)
+  if (title.match(/^[A-Z]/) && title.match(/[a-zA-Z0-9]$/))
+    state = "success"
+    description = "Commit title satisfies commit guidelines."
+  else
+    state = "failure"
+    description = "Commit title should begin with a capital letter and" \
+                  " does not end in a punctuation."
+  end
+else
+  state = "failure"
+  description = "Commit title should be less than 50 characters: actual #{title_len}"
+end
+
+# check commit message body length
+if (state == "success")
+  body.each_line.with_index() do |line, j|
+    line.chomp!
+#    puts "Line length is #{line.length}"
+    if (line.length >= 0 && line.length <= 72)
+      if (j == 0 && line.length != 0)
+	state = "failure"
+	description = "Commit message after title should be blank."
+      else
+        state = "success"
+	description = "Commit title and body lengths are within guidelines."
+      end
+    else
+      state = "failure"
+      description = "Commit body should be 72 characters or less per line: " \
+                    "line #{j+1} actual #{line.length}"
+      break
+    end
+  end
+end
+
+# check signed-off line
+if (state == "success")
+  lines = body.split(/\n+/)
+  if (lines.empty?)
+    state = "failure"
+    description = "Please provide a commit message body and sign off your commit."
```

Comment:
Lame comment...  The messages in this part of the script have become more polite.  Can you make the messages in this script consistent:  either all "terse" or all "polite", eh?